lib/internal/source_map/source_map_cache.js

```diff
@@ -45,30 +46,48 @@ const { fileURLToPath, pathToFileURL, URL, URLParse } = require('internal/url');
 let SourceMap;
 
 // This is configured with --enable-source-maps during pre-execution.
-let sourceMapsEnabled = false;
-function getSourceMapsEnabled() {
-  return sourceMapsEnabled;
+let sourceMapsSupport = ObjectFreeze({
+  __proto__: null,
+  enabled: false,
+  nodeModules: false,
+  generatedCode: false,
+});
+function getSourceMapsSupport() {
+  // Return a read-only object.
+  return sourceMapsSupport;
 }
 
 /**
  * Enables or disables source maps programmatically.
- * @param {boolean} val
+ * @param {boolean} enabled
+ * @param {object} options
+ * @param {boolean} [options.nodeModules]
+ * @param {boolean} [options.generatedCode]
  */
-function setSourceMapsEnabled(val) {
-  validateBoolean(val, 'val');
+function setSourceMapsSupport(enabled, options = kEmptyObject) {
+  validateBoolean(enabled, 'enabled');
+  validateObject(options, 'options');
+
+  const { nodeModules = false, generatedCode = false } = options;
```

Comment:
In the past, we always enabled the source map for nodeModules right? If so, having the default values to `false` won't be a breaking change? Maybe having this as `true` will be better if the my first assumption is right.

Reply:
Updated with `process.setSourceMapsEnabled(true)` implies `module.setSourceMapsSupport(true, { nodeModule: true, generatedCode: true })`.

The new API `module.setSourceMapsSupport` will still have the options default to `false`.